### PR TITLE
mech_calico: Ignore pre-live-migration messages

### PR DESCRIPTION
**Pull request by tj90241**
As of OpenStack Newton, an RPC call push was added to pre-
live-migration steps. Reference PRs:
https://review.opendev.org/c/openstack/nova/+/275073
https://review.openstack.org/#/c/260738/

This commit adds a guard to `mech_calico`'s port updating
logic so that it effectively ignores the pre-live-migration
notification referenced above.  Without this guard,
`mech_calico` mistakeningly thinks the port is becoming
unbound/disabled (and later enabled again) as part of the
pre-live-migration notifications and flaps the VM's port
accordingly.

In short, the addition of the guard is required to avoid
an unconditional, intermittent loss of connectivity for VMs
during the live-migration process.

Signed-off-by: Tyler Stachecki <tstachecki@bloomberg.net>

### CODE REVIEW

**Comment by nelljerram** [comment]
I've manually checked that I see these pre-migration updates in Tigera's own live migration testing, and am happy in principle that it will be OK (and in fact better) to skip them.

I've kicked off a full system test run - sadly not visible outside Tigera - at https://wavetank.tigera.io/view/Openstack/job/os-ms-fv-openstack-uussuri/43/ using https://launchpad.net/~project-calico/+archive/ubuntu/testing2, which was built from this PR.

File: networking_calico/plugins/ml2/drivers/calico/mech_calico.py [hunk 1]
```diff
@@ -773,14 +773,19 @@ def update_port_postcommit(self, context):
...
+            #   do nothing with right now.
             # - a port becoming bound (binding vif_type from unbound to bound);
             # - a port becoming unbound (binding vif_type from bound to
             #   unbound);
             # - an Icehouse migration (binding host id changed and port bound);
             # - an update (port bound at all times);
             # - a change to an unbound port (which we don't care about, because
             #   we do nothing with unbound ports).
-            if port_bound(port) and not port_bound(original):
+            if port['binding:profile'].get('migrating_to') is not None:
```
Comment (nelljerram) [inline networking_calico/plugins/ml2/drivers/calico/mech_calico.py]: Please can you add a `'binding:profile' in port` check here, to allow for older OpenStacks that don't have that key.  (And also our UTs!)

**Review by nelljerram — CHANGES_REQUESTED** [review]
Thanks @tj90241 .  This needs a change to allow for older OpenStacks, and I'm afraid it also now needs a rebase as well, following the merge of #36 , and then I'd like to run it through our system tests again.

Would you like to handle those change?  I'm happy to take over if you would prefer that.  (And I'll try to keep your name on the main commit!)

File: networking_calico/plugins/ml2/drivers/calico/mech_calico.py (hunk 1 above)
Comment (tj90241) [reply networking_calico/plugins/ml2/drivers/calico/mech_calico.py]: No problem -- done.  Did a`d.get('key', {})` to keep it all on one line.

**Review by nelljerram — APPROVED** [review]
Thanks, code change looks good.  Just want to run this now through our semaphore CI and (separate) ST suite; the latter is waiting for packages to be published at https://launchpad.net/~project-calico/+archive/ubuntu/testing2/+packages

**Comment by nelljerram** [comment]
Packages are ready so I've kicked off ST at https://wavetank.tigera.io/view/Openstack/job/os-ms-fv-openstack-uussuri/44/.  It takes a few hours so I'll check the results on Monday morning.

**Comment by nelljerram** [comment]
ST passed, so I will merge this change now.
